lib/websocket_test.go

```diff
@@ -26,23 +35,106 @@ func TestWebsocket(t *testing.T) {
 	instCtx, instCancel := context.WithCancel(context.Background())
 	defer instCancel()
 
-	inst, err := NewInstance(instCtx, tr.QriPath, OptConfig(cfg))
+	// create key store & add test key
+	kd := testkeys.GetKeyData(0)
+	ks, err := key.NewMemStore()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := ks.AddPubKey(context.Background(), kd.KeyID, kd.PrivKey.GetPublic()); err != nil {
+		t.Fatal(err)
+	}
+
+	// create instance
+	inst, err := NewInstance(instCtx, tr.QriPath, OptConfig(cfg), OptKeyStore(ks))
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	subsCount := inst.bus.NumSubscribers()
 
 	wsCtx, wsCancel := context.WithCancel(context.Background())
-	_, err = NewWebsocketHandler(wsCtx, inst)
+	defer wsCancel()
+
+	// create WebsocketHandler
+	websocketHandler, err := NewWebsocketHandler(wsCtx, inst)
 	if err != nil {
 		t.Fatal(err)
 	}
+	wsh := websocketHandler.(*wsHandler)
 
 	// websockets should subscribe the WS message handler
 	if inst.bus.NumSubscribers() != subsCount+1 {
 		t.Fatalf("failed to subscribe websocket handlers")
 	}
 
-	wsCancel()
+	// add connection
+	randIDStr := "test_connection_id_str"
+	SetIDRand(strings.NewReader(randIDStr))
+	connID := newID()
+	SetIDRand(strings.NewReader(randIDStr))
+
+	wsh.WSConnectionHandler(mockWebsocketWriterAndRequest())
```

Comment:
this tests that the `wsHandler` is setting up a connection & placing it in the `conns` map.

However, it doesn't set us up to be able to test the `wsHandler.read` or `wsHandler.handleMessage` methods. May need to expand on the `mockHijacker` struct below  & create methods that allow us to write on the connection.